### PR TITLE
test: Groups module e2e tests

### PR DESCRIPTION
**Pull request by glnro**
This PR fulfills the testing component for #1423 by introducing new e2e tests for the groups module.

CLI commands tested
-  `create-group` Create a group
-  `update-group-members` Add a group member
-  `update-group-members` Remove a group member
-  `create-group-policy` Create threshold decision policy
-  `create-group-policy` Create percentage decision policy
-  `submit-proposal` Submit MsgSend proposal
-  `vote` Vote on proposal
-  `exec` Execute proposal

REST endpoints
- [GroupMembers](https://docs.cosmos.network/v0.46/modules/group/05_client.html#groupmembers-2) `/cosmos/group/v1/group_members/{group_id}`
- [GroupInfo](https://docs.cosmos.network/v0.46/modules/group/05_client.html#groupinfo-2) `/cosmos/group/v1/group_info/{group_id}`
- [GroupsByAdmin](https://docs.cosmos.network/v0.46/modules/group/05_client.html#groupsbyadmin-2) `/cosmos/group/v1/groups_by_admin/{admin}`
- [GroupPoliciesByGroup](https://docs.cosmos.network/v0.46/modules/group/05_client.html#grouppoliciesbygroup-2) `/cosmos/group/v1/group_policies_by_group/{group_id}`
- [Proposal](https://docs.cosmos.network/v0.46/modules/group/05_client.html#proposal-3) `/cosmos/group/v1/proposal/{proposal_id}`
- [ProposalsByGroupPolicy](https://docs.cosmos.network/v0.46/modules/group/05_client.html#proposalsbygrouppolicy-2) `/cosmos/group/v1/proposals_by_group_policy/{address}`

User Flows
- Create Group
- Remove Group Member
- Create Threshold Decision Policy
- Create Percentage Decision Policy

- Submit MsgSend Proposal (via Threshold Decision Policy)
- Vote (Meet threshold)
- Exec Passed Proposal
- Verify Passed Proposal

- Submit MsgSend Proposal (via Percentage Decision Policy)
- Vote (Don't meet percentage)
- Verify Failed Proposal

This test covers the basic user stories for using `x/group`. There's a fair bit of refactoring that can be done to these tests and there remain a few test cases that can be added for a more comprehensive groups test suite.

### CODE REVIEW

File: tests/e2e/e2e_groups_test.go
Comment (okwme) [inline tests/e2e/e2e_groups_test.go]: feels like this logic should live inside this method. it's not used by any other function and it changes the values of global variables. For instance I'd expect line 80 (`len(members)`) to be 0 based on the variable being initialized above. If the reader is just expected to jump down to the `prepareGroupFiles` definition then it hasn't actually improved legibility of the code by putting it into a separate function but actually made it harder to follow. Another way could be to make it a functional method that doesn't change state itself but returns the file and returns `members` that are then re-assigned in line.

this is low priority feedback and I wouldn't call it essential to be changed for this PR to be merged but still hopefully valuable feedback from perspective of someone reading tests without the context you built up by writing them.

File: tests/e2e/e2e_groups_test.go
Comment (okwme) [inline tests/e2e/e2e_groups_test.go]: why submit the votes to different validators?

File: tests/e2e/e2e_util_test.go
Comment (okwme) [inline tests/e2e/e2e_util_test.go:619]: is there no `--from` flag for these commands?

File: tests/e2e/e2e_groups_test.go
Comment (okwme) [inline tests/e2e/e2e_groups_test.go]:
```suggestion
	s.executeVoteGroupProposal(s.chainA, 1, chainAAPIEndpoint, strconv.Itoa(proposalId), aliceAddr, group.VOTE_OPTION_ABSTAIN.String(), "Alice votes abstain")
```

File: tests/e2e/e2e_groups_test.go
Comment (okwme) [reply tests/e2e/e2e_groups_test.go]:
```suggestion
	s.executeVoteGroupProposal(s.chainA, 1, chainAAPIEndpoint, strconv.Itoa(proposalId), aliceAddr, group.VOTE_OPTION_YES.String(), "Alice votes yes")
```

File: tests/e2e/e2e_setup_test.go
Comment (okwme) [inline tests/e2e/e2e_setup_test.go:97]: should these types be imported from the SDK?

**Review by okwme — APPROVED** [review]
Looks good!
curious to hear your response to some of the questions if you'd like to change anything but nothing is a requirement from my pov.
feel free to merge at your discretion @glnro

**Comment by okwme** [comment]
actually after speaking with yaru let's wait until the global fee work is merged in.
it will be easier to change this PR to accommodate those changes than the other way around

File: tests/e2e/e2e_setup_test.go
Comment (glnro) [reply tests/e2e/e2e_setup_test.go:97]: yeah todo as a second refactor pass

File: tests/e2e/e2e_groups_test.go
Comment (glnro) [reply tests/e2e/e2e_groups_test.go]: Because one validator is the admin, and one validator is a group member. I've had issues signing transactions otherwise like submitting a proposal or voting with the accounts created in the test. It's another thing to refactor.

File: tests/e2e/e2e_groups_test.go
Comment (mmulji-ic) [inline tests/e2e/e2e_groups_test.go]: /root/.gaia/config/%s should be a const

Comment (glnro) [reply tests/e2e/e2e_groups_test.go]: I think actually this might be the problem, I'm going to confirm what the data dir actually is

Comment (mmulji-ic) [reply tests/e2e/e2e_groups_test.go]: I slacked you earlier ... I've committed to this branch already the fix
swap /root/.gaia/config/%s  for /home/nonroot/.gaia/config/%s

File: tests/e2e/e2e_util_test.go
Comment (glnro) [reply tests/e2e/e2e_util_test.go:619]: no `gaiad tx group vote [proposal-id] [voter] [vote-option] [metadata] [flags]` it's assumed the voter is the `from` address

File: tests/e2e/e2e_groups_test.go
Comment (glnro) [reply tests/e2e/e2e_groups_test.go]: Yeah I can see how that could be confusing...outside of writing an additional function that just gets the length of the original list of members I don't know that putting that setup in the body of the main test helps with legibility but it's worth a discussion when we set aside time to do the refactor.

File: tests/e2e/e2e_groups_test.go
Comment (glnro) [reply tests/e2e/e2e_groups_test.go]: 👍

**Pull request merged**
